### ISSUE TITLE
New user registration is broken

### ISSUE THREAD

**Issue by IlyaSemenov** [Bannerets/tdl#62]
New user registration is no longer working since a few days ago.

In particular, this code:

https://github.com/Bannerets/tdl/blob/95e0076408c3168a375ac7a2dcf9e48ea239918d/packages/tdl/src/client.js#L447-L467

leads to this error:

```json
{
    "_": "error",
    "code": 400,
    "message": "PHONE_NUMBER_UNOCCUPIED",
    "@extra": null
}
```

This API has apparently been deprecated and now disabled. The clients are supposed to handle `authorizationStateWaitCode` and `authorizationStateWaitRegistration` separately (first one for code only, second one for name only). See the official example:

https://github.com/tdlib/td/blob/b1347c32262d78e28f893b104b493aa2315987d6/example/python/tdjson_example.py#L127-L131

I made a quick ad-hoc patch (for tdl v.5 which I'm still using) and that worked.

**Comment by Bannerets** [comment]
Yes, the `login` function doesn't work on TDLib v1.5.0 (and above). `tdl` hasn't been updated since 1.5.0 release yet.

**Comment by IlyaSemenov** [comment]
That's how it's supposed to work now:

```js
      case 'authorizationStateWaitCode':
        {
          const loginDetails = this._authRequired();
          if (loginDetails.type !== 'user') return;

          const code = await loginDetails.getAuthCode(false);
          return this._send({
            _: 'checkAuthenticationCode',
            code
          });
        }

      case 'authorizationStateWaitRegistration':
        {
          const loginDetails = this._authRequired();
          if (loginDetails.type !== 'user') return;

          const { firstName, lastName = '' } = await loginDetails.getName();
          return this._send({
            _: 'registerUser',
            first_name: firstName,
            last_name: lastName
          });
        }
```

**Comment by IlyaSemenov** [comment]
The problem is, it's broken even for older releases. I was using tdl 5 with tdlib 1.2 (from https://github.com/Bannerets/tdlib-binaries/tree/master/1.2.0/macos) and it just stopped working.

**Comment by Bannerets** [comment]
> it's broken even for older release

Hmm, that's strange. `authorizationStateWaitRegistration` is available only since `v1.5.0`, so the problem doesn't look fixable for older versions.
The code above also uses added in 1.5.0 `authorizationStateWaitRegistration` and `registerUser`, thus it shouldn't work on lower versions. (see https://github.com/tdlib/td/blob/v1.5.0/CHANGELOG.md)

**Comment by IlyaSemenov** [comment]
That's probably so. (I didn't test the new additions with 1.2.)
